runtime/src/tests/work-item/work-item-list/labels.spec.js

```diff
@@ -0,0 +1,132 @@
+  var firstitem = {
+    'id': 'id0',
+  },
+  defaultNumberOfLabels = 5,  // From mock data
+  newLabelTitle = "My Test Label",
+  defaultAttchedLabelIndex = 2; // From mock data
+
+  beforeEach(function () {
+    testSupport.setBrowserMode('desktop');
+    page = new WorkItemListPage(true);
+    testSupport.setTestSpace(page);
+  });
+
+  it('Verify Add label button exists', function(){
+    detailPage = page.clickWorkItemTitle(page.firstWorkItem, firstitem.id);
+    expect(detailPage.addLabelButton.isPresent()).toBeTruthy();
+  });
+
+  it('Verify Add label button is clickable', function(){
+    detailPage = page.clickWorkItemTitle(page.firstWorkItem, firstitem.id);
+    expect(detailPage.selectLabelDropdown.isPresent()).toBeFalsy();
+    detailPage.clickAddLabelButton();
+    expect(detailPage.selectLabelDropdown.isPresent()).toBeTruthy();
+  });
+
+  it('Verify Create New Label Button Exists', function(){
+    detailPage = page.clickWorkItemTitle(page.firstWorkItem, firstitem.id);
+    detailPage.clickAddLabelButton();
+    expect(detailPage.createLabelButton.isPresent()).toBeTruthy();
+  });
+
+  it('Verify Create New Label Button is clickable', function(){
+    detailPage = page.clickWorkItemTitle(page.firstWorkItem, firstitem.id);
+    detailPage.clickAddLabelButton();
+    expect(detailPage.labelInputDiv.isPresent()).toBeTruthy();
+  });
+
+  it('Verify Create New Label', function(){
+    detailPage = page.clickWorkItemTitle(page.firstWorkItem, firstitem.id);
+    detailPage.clickAddLabelButton();
+    let origLabelCount
+    detailPage.labelsCount.then(function(count){
+      origLabelCount = count
+    });
+    detailPage.clickCreateLabelButton();
+    detailPage.setLabelName(newLabelTitle);
+    detailPage.clickLabelCheck();
+    // Verify label count has increased by 1
+    detailPage.labelsCount.then(function(count){
+      expect(count).toBe(origLabelCount + 1);
+    });
+    // Verify label exists in the list
+    expect(detailPage.listOfLabels().getText()).toContain(detailPage.labelTitleFromIndex(1));
+  })
+
+  it('Verify adding existing labels', function(){
+    detailPage = page.clickWorkItemTitle(page.firstWorkItem,firstitem.id);
+    detailPage.clickAddLabelButton();
+    detailPage.selectLabel(1);
+    detailPage.selectLabel(3);
+
+    expect(detailPage.attachedLabels().getText()).toContain(detailPage.labelTitleFromIndex(3));
+    expect(detailPage.attachedLabels().getText()).toContain(detailPage.labelTitleFromIndex(1));
+    detailPage.clickLabelClose();
+  });
+
+  it('Verify removing existing label', function(){
+    detailPage = page.clickWorkItemTitle(page.firstWorkItem,firstitem.id);
+    detailPage.clickAddLabelButton();
+    detailPage.selectLabel(defaultAttchedLabelIndex);
+    detailPage.clickLabelClose();
+
+    // Verify Label is removed (in detail page)
+    expect(detailPage.labelTitleFromIndex(defaultAttchedLabelIndex)).toBe("");
+  });
+
+  it('Verify adding new labels', function(){
```

Comment:
Eliminate the next test and check for label availability in the list view in this test itself. This makes the tests faster and eliminated redundancy.